apps/condo/migrations/20240923115003-0426_newsitemtemplate_category_newsitemtemplate_name_and_more.js

```diff
@@ -0,0 +1,55 @@
+// auto generated by kmigrator
+// KMIGRATOR:0426_newsitemtemplate_category_newsitemtemplate_name_and_more:IyBHZW5lcmF0ZWQgYnkgRGphbmdvIDUuMC43IG9uIDIwMjQtMDktMjMgMDY6NTAKCmZyb20gZGphbmdvLmRiIGltcG9ydCBtaWdyYXRpb25zLCBtb2RlbHMKCgpjbGFzcyBNaWdyYXRpb24obWlncmF0aW9ucy5NaWdyYXRpb24pOgoKICAgIGRlcGVuZGVuY2llcyA9IFsKICAgICAgICAoJ19kamFuZ29fc2NoZW1hJywgJzA0MjVfbWVzc2FnZV9tZXNzYWdlX3VzZXJfc3RhdHVzX3NlbnRhdCcpLAogICAgXQoKICAgIG9wZXJhdGlvbnMgPSBbCiAgICAgICAgbWlncmF0aW9ucy5BZGRGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0nbmV3c2l0ZW10ZW1wbGF0ZScsCiAgICAgICAgICAgIG5hbWU9J2NhdGVnb3J5JywKICAgICAgICAgICAgZmllbGQ9bW9kZWxzLkNoYXJGaWVsZChibGFuaz1UcnVlLCBjaG9pY2VzPVsoJ3dhdGVyJywgJ3dhdGVyJyksICgnaGVhdGluZycsICdoZWF0aW5nJyksICgnZWxlY3RyaWNpdHknLCAnZWxlY3RyaWNpdHknKSwgKCdhbGxVdGlsaXRpZXMnLCAnYWxsVXRpbGl0aWVzJyksICgnZWxlY3RyaWNpdHlBbmRXYXRlcicsICdlbGVjdHJpY2l0eUFuZFdhdGVyJyksICgnbGlnaHQnLCAnbGlnaHQnKSwgKCdlbGV2YXRvcnMnLCAnZWxldmF0b3JzJyksICgndGVsZXBob25lJywgJ3RlbGVwaG9uZScpLCAoJ21ldGVycycsICdtZXRlcnMnKSwgKCdnYXMnLCAnZ2FzJyksICgnaW50ZXJjb20nLCAnaW50ZXJjb20nKSwgKCdjbGVhbmluZycsICdjbGVhbmluZycpLCAoJ3Nub3cnLCAnc25vdycpLCAoJ290aGVyJywgJ290aGVyJyldLCBtYXhfbGVuZ3RoPTUwLCBudWxsPVRydWUpLAogICAgICAgICksCiAgICAgICAgbWlncmF0aW9ucy5BZGRGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0nbmV3c2l0ZW10ZW1wbGF0ZScsCiAgICAgICAgICAgIG5hbWU9J25hbWUnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuVGV4dEZpZWxkKGRlZmF1bHQ9J1RlbXAnKSwKICAgICAgICAgICAgcHJlc2VydmVfZGVmYXVsdD1GYWxzZSwKICAgICAgICApLAogICAgICAgIG1pZ3JhdGlvbnMuQWRkRmllbGQoCiAgICAgICAgICAgIG1vZGVsX25hbWU9J25ld3NpdGVtdGVtcGxhdGVoaXN0b3J5cmVjb3JkJywKICAgICAgICAgICAgbmFtZT0nY2F0ZWdvcnknLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuVGV4dEZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgICAgICBtaWdyYXRpb25zLkFkZEZpZWxkKAogICAgICAgICAgICBtb2RlbF9uYW1lPSduZXdzaXRlbXRlbXBsYXRlaGlzdG9yeXJlY29yZCcsCiAgICAgICAgICAgIG5hbWU9J25hbWUnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuVGV4dEZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgIF0K
+
+exports.up = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Add field category to newsitemtemplate
+--
+ALTER TABLE "NewsItemTemplate" ADD COLUMN "category" varchar(50) NULL;
+--
+-- Add field name to newsitemtemplate
+--
+ALTER TABLE "NewsItemTemplate" ADD COLUMN "name" text DEFAULT 'Temp' NOT NULL;
+ALTER TABLE "NewsItemTemplate" ALTER COLUMN "name" DROP DEFAULT;
+--
+-- [CUSTOM] Set "name" = "title" in old records
+--
+UPDATE "NewsItemTemplate" SET name = title WHERE name = 'Temp';
```

Comment:
what's going on here? You just added a "name" field

Reply:
Because "temp" is the default temporary value for old records in the database. We want to set all old records to name = title

Reply:
So why do you need this `temp` value?
For you, old templates are all those created before the release.
That is, all records are old for you and all need to be renamed. Isn't that right?

Reply:
I need to add another column to the database, which should be mandatory. Also for the old values ​​I need the value name = title, i.e. it is dynamic. It is possible to set such a value only after the static value ("Temp") is set, and then updated to a new one by overwriting the temporary value and setting a new, dynamic value.

Is this bad? Are there any other options? :-)

Reply:
Ok, now I get it :)

Please write this in the comments to the migration

Reply:
Can you not use an UPDATE? I'm personally OK with this (you also do not really need to filter on 'Temp') but maybe there are more elegant solution

Reply:
I will leave this solution because I don't know any other way to set a dynamic value except UPDATE. I will also leave the filtering by "Temp" because I don't think it can do any harm.